packages/color-scale-block/src/ColorScaleBlock.tsx

```diff
@@ -0,0 +1,520 @@
+    const draggingId = useRef<number | null | undefined>();
+    const dragStartPos = useRef<number | null | undefined>();
+    const dragStartWidth = useRef<number | null | undefined>();
+    const lastDragPos = useRef<number | null | undefined>();
+    const timerToUpdateBlockSettings = useRef<ReturnType<typeof setTimeout>>();
+
+    const deleteColor = (id: number) => {
+        if (!id) {
+            return;
+        }
+
+        const reorderedList = displayableItems?.filter((item) => item.id !== id);
+        setBlockSettings({ ...blockSettings, colorInput: reorderedList });
+    };
+
+    const calculateLeftPosition = (index: number, width?: number) => {
+        let leftPos = 0;
+        const defaultWidth = width ?? DEFAULT_COLOR_SQUARE_WIDTH;
+        displayableItems?.map((color: ColorProps, loopIndex: number) => {
+            if (loopIndex < index) {
+                leftPos += color?.width ? color.width : defaultWidth;
+            }
+            return color;
+        });
+        return leftPos;
+    };
+
+    const updateColor = (newColor: ColorProps) => {
+        if (displayableItems.findIndex((item) => item.id === newColor.id) > -1) {
+            return;
+        }
+
+        const colorScaleBlockWidth = colorScaleBlockRef?.current?.getBoundingClientRect().width;
+
+        let newColors = [...displayableItems, newColor];
+
+        if (newColors.length === 2 && colorScaleBlockWidth && !newColors[1].width) {
+            newColors[0].width = colorScaleBlockWidth / 2;
+            newColors[1].width = colorScaleBlockWidth / 2;
+        } else {
+            newColors = calculateWidths(newColors, colorScaleBlockRef, true);
+        }
+
+        setBlockSettings({
+            ...blockSettings,
+            colorInput: newColors,
+        });
+    };
+
+    useEffect(() => {
+        setColorPickerPalettes(
+            appBridgePalettes.map((palette) => {
+                const colors = palette.colors.map((color) => {
+                    return {
+                        alpha: color.alpha ? color.alpha / 255 : 1,
+                        red: color.red ?? 0,
+                        green: color.green ?? 0,
+                        blue: color.blue ?? 0,
+                        name: color.name ?? '',
+                    };
+                });
+
+                return {
+                    id: palette.id,
+                    title: palette.name,
+                    colors,
+                };
+            })
+        );
+    }, [appBridgePalettes, appBridge]);
+
+    useEffect(() => {
+        if (blockSettings?.colorInput) {
+            setDisplayableItems(blockSettings.colorInput);
+        }
+        const currentHeight = blockSettings.customHeight ? blockSettings.heightInput : blockSettings.heightSlider;
+        if (colorScaleHeight !== currentHeight) {
+            setColorScaleHeight(currentHeight);
+        }
+        if (colorScaleBlockRef?.current) {
```

Comment:
Fix this nested ifs. You can make it cleaner and more readable.

Reply:
Is this better?

```
        if (
            colorScaleBlockRef.current &&
            blockSettings.colorInput &&
            blockSettings.colorInput.length >= MINIMUM_NUMBER_OF_COLORS
        ) {
            const needToCalculateWidths = blockSettings.colorInput.find((color) => !color.width);

            if (needToCalculateWidths) {
                const colorsWithNewWidths = calculateWidths(blockSettings.colorInput, colorScaleBlockRef, false);

                setBlockSettings({
                    ...blockSettings,
                    colorInput: colorsWithNewWidths,
                });
            }
        }
```

Reply:
```  
const hasColors = !colorScaleBlockRef.current && blockSettings.colorInput?.length >= MINIMUM_NUMBER_OF_COLORS
const needToCalculateWidths =  blockSettings.colorInput.find((color) => !color.width);
  
if (needToCalculateWidths && hasColors) {
    const colorsWithNewWidths = calculateWidths(blockSettings.colorInput, colorScaleBlockRef, false);

    setBlockSettings({
        ...blockSettings,
        colorInput: colorsWithNewWidths,
    });
}
```